src/plugins/explore/public/helpers/data_table_helper.tsx

```diff
@@ -82,11 +86,11 @@ export function getTimeColumn(
  */
 export function getLegacyDisplayedColumns(
   columns: string[],
-  indexPattern: IndexPattern,
+  indexPattern: IndexPattern | Dataset,
   hideTimeField: boolean,
   isShortDots: boolean
 ): LegacyDisplayedColumn[] {
-  if (!Array.isArray(columns) || typeof indexPattern !== 'object' || !indexPattern.getFieldByName) {
+  if (!Array.isArray(columns) || !indexPattern || !indexPattern.getFieldByName) {
     return [];
   }
   // TODO: Remove overrides once we support PPL/SQL sorting
```

Comment:
im not sure if this got ported over but i think this might bite us in the future. ie what if the field is not searchable or aggregatable